### PR TITLE
[Xtensa] 8-bit arithmetic improvements + some other smaller changes

### PR DESCRIPTION
**Pull request by Aelphy**
Changes from Cadence

### CODE REVIEW

File: src/XtensaOptimize.cpp [hunk 1]
```diff
@@ -937,8 +938,11 @@ class MatchXtensaPatterns : public IRGraphMutator {
             {"halide_xtensa_narrow_i48_with_rounding_shift_i16", i16(rounding_shift_right(i32(wild_i48x), wild_i32))},
             {"halide_xtensa_narrow_i48_with_rounding_shift_u16", u16(rounding_shift_right(u32(wild_i48x), wild_u32))},
 
+#if 0  // Somehow this pattern is preventing a i16(i32(i48x) >> wild_i32) to be recognized as a PACKVRNRNX48
```
Comment (steven-johnson) [inline src/XtensaOptimize.cpp:941]: Do we need to open an Issue for this?

Comment (vksnk) [reply src/XtensaOptimize.cpp:941]: I think I also ran into this issue recently and have a fix.

**Comment by steven-johnson** [comment]
Ready to land or is the `if 0` going to be dealt with first?

File: src/CodeGen_Xtensa.cpp [hunk 2]
```diff
@@ -4057,6 +4105,10 @@ void CodeGen_Xtensa::visit(const Allocate *op) {
             }
         }
 
+        if (op->type.bits() == 24) {
```
Comment (vksnk) [inline src/CodeGen_Xtensa.cpp:4108]: This doesn't look right, looks like workaround for something and should be done in schedule instead.

File: src/XtensaOptimize.cpp (hunk 1 above)
Comment (vksnk) [reply src/XtensaOptimize.cpp:941]: This needs to be resolved before submission, I'll double check if it's the same issue I had.

**Comment by steven-johnson** [comment]
Where does this PR stand?

**Comment by steven-johnson** [comment]
Monday Morning Review Ping

File: src/XtensaOptimize.cpp (hunk 1 above)
Comment (vksnk) [reply src/XtensaOptimize.cpp:938]: Resolved now, but you'll need to sync.

**Comment by Aelphy** [comment]
Seems ready from my side

**Review by vksnk — APPROVED** [review]

**Pull request merged**